### PR TITLE
[osgearth] Make all find_packages required in effort to make this port less "flaky", and turn on in CI.

### PR DESCRIPTION
**Pull request by BillyONeal**
Related: https://github.com/microsoft/vcpkg/pull/13423

Also adds "static-crt" identifier to platform expressions.

### CODE REVIEW

File: toolsrc/src/vcpkg/platform-expression.cpp [hunk 1]
```diff
@@ -51,6 +52,7 @@ namespace vcpkg::PlatformExpression
             {"emscripten", Identifier::emscripten},
             {"ios", Identifier::ios},
             {"static", Identifier::static_link},
+            {"static-crt", Identifier::static_link},
```
Comment (ras0219) [inline toolsrc/src/vcpkg/platform-expression.cpp:55]:
```suggestion
            {"staticcrt", Identifier::static_link},
```

File: ports/osgearth/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "osgearth",
+  "version-string": "3.1",
```
Comment (ras0219) [inline ports/osgearth/vcpkg.json:3]:
```suggestion
  "version": "3.1",
```

**Review by ras0219 — APPROVED** [review]
The tool changes LGTM.

File: toolsrc/src/vcpkg/platform-expression.cpp (hunk 1 above)
Comment (BillyONeal) [reply toolsrc/src/vcpkg/platform-expression.cpp:55]: I assume you want me to back out the identifier parsing change as well here; should we bikeshed this?

**Comment by Neumann-A** [comment]
https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html for reference ;)

**Comment by BillyONeal** [comment]
> https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html for reference ;)

This is doing the opposite though (forcing them to be REQUIRED)

**Comment by Neumann-A** [comment]
>This is doing the opposite though (forcing them to be REQUIRED)

This should only be required if they are not added to build-depends field. Otherwise disabling should be the preferred option. Only the ones already having an option should probably have the REQUIRED keyword. 
Otherwise for debugging patch in: https://cmake.org/cmake/help/latest/module/FeatureSummary.html to get an overview what was found and what not.

**Comment by BillyONeal** [comment]
@Neumann-A I added all the associated find_package'd things to dependencies, does that resolve your concern?

**Comment by Neumann-A** [comment]
> I added all the associated find_package'd things to dependencies, does that resolve your concern?

Yes

**Pull request merged**